pkg/registry/core/service/allocator/storage/storage.go

```diff
@@ -80,17 +80,12 @@ func NewEtcd(alloc allocator.Snapshottable, baseKey string, resource schema.Grou
 	}, nil
 }
 
-// Allocate attempts to allocate the item locally and then in etcd.
+// Allocate attempts to allocate the item.
 func (e *Etcd) Allocate(offset int) (bool, error) {
 	e.lock.Lock()
 	defer e.lock.Unlock()
 
-	ok, err := e.alloc.Allocate(offset)
-	if !ok || err != nil {
-		return ok, err
-	}
-
-	err = e.tryUpdate(func() error {
+	err := e.tryUpdate(func() error {
```

Comment:
Just putting my scalability hat for now, I'm not sure I understand what does it change from scalability/performance prespective. We get+write from etcd both before and after.

So, what is the difference? That in case the IP was already allocated, we don't do get? [And that has a possibility to race?]
If that's the case, we should fix it no matter what scalability impact it has...

Reply:
yeah, I think that's the only change here on that regard. Thanks for confirming

Reply:
that is how I read it too